sklearn/tree/_criterion.pyx

```diff
@@ -309,24 +310,18 @@ cdef class ClassificationCriterion(Criterion):
         self.end = end
         self.n_node_samples = end - start
         self.weighted_n_samples = weighted_n_samples
-        cdef double weighted_n_node_samples = 0.0
+        self.weighted_n_node_samples = 0.0
 
-        cdef SIZE_t n_outputs = self.n_outputs
-        cdef SIZE_t* n_classes = self.n_classes
-        cdef SIZE_t label_count_stride = self.label_count_stride
-        cdef double* label_count_total = self.label_count_total
+        cdef double* node_sum = self.node_sum
 
-        cdef SIZE_t i = 0
-        cdef SIZE_t p = 0
-        cdef SIZE_t k = 0
-        cdef SIZE_t c = 0
+        cdef SIZE_t i, p, k, c
```

Comment:
I understand that we want to put each variable on a different line, but in the case of single character iterators it seems a waste of space.